### PR TITLE
Issue #13345: EnableUnnecessaryParenthesesCheckExamplesTest

### PR DESCRIPTION
**Pull request by mahfouz72**
issue #13345: 
this PR enable example tests for UnnecessaryParenthesesCheckExamplesTest
mvn verify passed

### CODE REVIEW

**Comment by mahfouz72** [comment]
@romani  I want to remove violation messages to not exceed the line limit(85 char) but I got this error 
The violation message should be specified 
and I don't know why I see a lot of PR that doesn't specify the violation message
but to solve this I used a short message for all violations (// violation 'Unnecessary parentheses') is it ok or do you have another suggestion?

**Comment by romani** [comment]
Search in code phrases ` violation above` ,` violation below` to see solutions

**Comment by mahfouz72** [comment]
> Search in code phrases ` violation above` ,` violation below` to see solutions

done with using violation below messages

**Comment by mahfouz72** [comment]
Github, generate site

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/Example1.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    <module name="UnnecessaryParentheses"/>
+  </module>
+</module>
+*/
+package com.puppycrawl.tools.checkstyle.checks.coding.unnecessaryparentheses;
+
+import java.util.List;
+
+// xdoc section -- start
+class Example1 {
```
Comment (nrmancuso) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/Example1.java:13]:
```suggestion
class Example1 {

```

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/Example1.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  public int square(int a, int b) {
+    // violation below, 'Unnecessary parentheses around assignment right-hand side'
+    int square = (a * b);
+    // violation below, 'Unnecessary parentheses around identifier 'square''
+    return (square);
+  }
+  int sumOfSquares = 0;
```
Comment (nrmancuso) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/Example1.java:20]:
```suggestion

  int sumOfSquares = 0;
```

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/Example1.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  public void sumOfSquares() {
+    // violation below, 'Unnecessary parentheses around literal '0''
+    for (int i = (0); i < 10; i++) {
+      // violation below, 'Unnecessary parentheses around assignment right-hand side'
+      int x = (i + 1);
+      sumOfSquares += (square(x,x));  // 2 violations
+    }
+  }
+  List<String> myList = List.of("a1", "b1", "c1");
```
Comment (nrmancuso) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/Example1.java:29]:
```suggestion

  List<String> myList = List.of("a1", "b1", "c1");
```

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/Example1.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  public void filter() {
+    myList.stream()
+        // violation below, 'Unnecessary parentheses around lambda value'
+        .filter((s) -> s.startsWith("c"))
+            .forEach(System.out::println);
+  }
+  int a = 10, b = 12, c = 15;
```
Comment (nrmancuso) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/Example1.java:36]:
```suggestion

  int a = 10, b = 12, c = 15;
```

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/Example1.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+                // violation below, 'Unnecessary parentheses around expression'
+                || (c >= 3 && a <= 7)) {
+      return;
+    }
+    // violation below, 'Unnecessary parentheses around expression'
+    if ((-a) != -27
+            && b > 5) {
+      return;
+    }
+    if (x==(a <= 15)) {
```
Comment (nrmancuso) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/Example1.java:52]:
```suggestion
    if (x == (a <= 15)) {
```

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/Example1.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+      return;
+    }
+    if (x==(y == z)) {
```
Comment (nrmancuso) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/Example1.java:55]:
```suggestion
    if (x == (y == z)) {
```

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/Example2.java [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+*/
+package com.puppycrawl.tools.checkstyle.checks.coding.unnecessaryparentheses;
+
+// xdoc section -- start
+class Example2 {
+
+  void method() {
+    int x = 9, y = 8;
+    // violation below, 'Unnecessary parentheses around expression'
+    if(x>= 0 ^ (x<=8 & y<=11)
```
Comment (nrmancuso) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/Example2.java:18]:
```suggestion
    if (x>= 0 ^ (x<=8 & y<=11)
```

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/Example2.java [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+         ^ y>=8) {
```
Comment (nrmancuso) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/Example2.java:19]:
```suggestion
         ^ y >= 8) {
```

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/Example2.java [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+      return;
+    }
+    if(x>= 0 ^ x<=8 & y<=11 ^ y>=8) {
```
Comment (nrmancuso) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/Example2.java:22]:
```suggestion
    if (x >= 0 ^ x <= 8 & y <= 11 ^ y >= 8) {
```

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/Example2.java [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+      return;
+    }
+    // violation below, 'Unnecessary parentheses around expression'
+    if(x>= 0 || (x<=8 & y<=11)
```
Comment (nrmancuso) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/Example2.java:26]:
```suggestion
    if ( x>= 0 || (x <= 8 & y <=11)
```

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/Example2.java [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+        && y>=8) {
```
Comment (nrmancuso) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/Example2.java:27]:
```suggestion
        && y >= 8) {
```

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/Example2.java [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+      return;
+    }
+    if(x>= 0 || x<=8 & y<=11 && y>=8) {
```
Comment (nrmancuso) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/Example2.java:30]:
```suggestion
    if (x >= 0 || x <= 8 & y <= 11 && y >= 8) {
```

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/Example2.java [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+      return;
+    }
+    if(x>= 0 & (x<=8 ^ y<=11) & y>=8) {
```
Comment (nrmancuso) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/Example2.java:33]:
```suggestion
    if (x >= 0 & (x <= 8 ^ y <= 11) & y >= 8) {
```

**Review by nrmancuso — CHANGES_REQUESTED** [review]
Items:

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/Example2.java (hunk 13 above)
Comment (mahfouz72) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/Example2.java]: @nrmancuso done

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/Example1.java (hunk 1 above)
Comment (mahfouz72) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/Example1.java:13]: @nrmancuso these changes make the example file length more than 60 lines.so I didn't apply them cause this is a violation `Example1.java:1: File length is 64 lines (max allowed is 60). [FileLength]` So please If you have another solution for this make a review for it

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/Example1.java (hunk 5 above)
Comment (mahfouz72) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/Example1.java]: done

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/Example2.java (hunk 9 above)
Comment (mahfouz72) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/Example2.java]: done

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/Example2.java (hunk 10 above)
Comment (mahfouz72) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/Example2.java]: done

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/Example2.java (hunk 11 above)
Comment (mahfouz72) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/Example2.java]: done

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/Example2.java (hunk 12 above)
Comment (mahfouz72) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/Example2.java]: done

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/Example1.java (hunk 1 above)
Comment (nrmancuso) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/Example1.java:13]: @mahfouz72 thanks for bringing this up. I personally find this example to be hard to follow with no empty lines to separate the methods. @rnveach thoughts on extending file length limit for examples?

**Comment by nrmancuso** [comment]
Github, generate site

**Comment by mahfouz72** [comment]
Github, generate site

**Comment by mahfouz72** [comment]
@nrmancuso please check these latest changes. I tried to maintain the spaces to improve the example readability as you requested. and also didn't exceed the file length limit to avoid the violation. please take a look at it and tell me if you have any other suggestion

**Review by romani — APPROVED** [review]
Ok to merge

**Review by nrmancuso — APPROVED** [review]
